src/Workspaces/Core/Portable/Workspace/ProjectSystem/FileWatchedPortableExecutableReferenceFactory.cs

```diff
@@ -101,34 +101,68 @@ internal sealed class FileWatchedPortableExecutableReferenceFactory
 
     public event EventHandler<string>? ReferenceChanged;
 
-    public PortableExecutableReference CreateReferenceAndStartWatchingFile(string fullFilePath, MetadataReferenceProperties properties)
+    /// <summary>
+    /// Gets or creates a PortableExecutableReference instance for the given file path and properties.
+    /// </summary>
+    public PortableExecutableReference CreateReference(string fullFilePath, MetadataReferenceProperties properties)
```

Comment:
creation needs to be separate from watching in case the transformation (where the reference is created) fails.